### PR TITLE
fix: [nuxt-3] Gallery Item List and Transfer NFT issues 

### LINKED ISSUE #7321: Floor button not working in mass list
- floor button selector does not work when clicking it for the first time, after you click the +5% etc., then it works
- 

https://github.com/kodadot/nft-gallery/assets/90852205/139b1a74-5515-41c6-99bd-79da4b93e4aa

image

### PR DESCRIPTION
**Pull request by hassnian**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- Make list and transfer work gallery item 

- [x] Ref #7403
- [x] Closes #7321 

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; 

transfer

CleanShot 2023-10-03 at 11 23 18

list 

CleanShot 2023-10-03 at 12 15 27

### <samp>🤖 Generated by Copilot at 12ceb09</samp>

This pull request refactors some components to use Vue 3 and Vue Router 4 features, fixes some minor UI issues, and renames and moves the `ListingCartModal` component to `ItemTransferModal` to better reflect its functionality.

### <samp>🤖 Generated by Copilot at 12ceb09</samp>

> _`ItemTransferModal`_
> _New name, new features, new home_
> _Autumn of changes_

### CODE REVIEW

File: components/common/itemTransfer/ItemTransferModal.vue [hunk 1]
```diff
@@ -13,7 +13,7 @@
             variant="text"
             no-shadow
             icon="xmark"
-            icon-pack="fa-sharp"
+            icon-pack="fasr"
```
Comment (preschian) [inline components/common/itemTransfer/ItemTransferModal.vue:16]: `fasr` is already default value

**Review by roiLeo — APPROVED** [review]

File: components/common/itemTransfer/ItemTransferModal.vue [hunk 2]
```diff
@@ -169,7 +169,7 @@ const transfer = () => {
     interaction: Interaction.SEND,
     urlPrefix: urlPrefix.value,
     address: address.value,
-    tokenId: $route.params.id,
+    tokenId: route.params.id,
```
Comment (preschian) [inline components/common/itemTransfer/ItemTransferModal.vue:172]: can we fix this?

```
Type 'string | string[]' is not assignable to type 'string'.
  Type 'string[]' is not assignable to type 'string'.ts(2322)
```

File: components/common/listingCart/shared/ListingCartPriceInput.vue [hunk 3]
```diff
@@ -14,6 +14,7 @@
       v-if="check"
       no-shadow
       class="shade-border-color ml-2 height-40"
+      icon-pack="fas"
```
Comment (preschian) [inline components/common/listingCart/shared/ListingCartPriceInput.vue:17]: can we use `fass` in here? if not, keep it `fas`

**Review by preschian — APPROVED** [review]
otherwise lgtm

File: components/common/listingCart/shared/ListingCartPriceInput.vue (hunk 3 above)
Comment (hassnian) [reply components/common/listingCart/shared/ListingCartPriceInput.vue:17]: will keep fas

**Review by prury — APPROVED** [review]

**Comment by hassnian** [comment]
@yangwao psyout thanks

**Pull request merged**